crates/ruff_workspace/src/options.rs

```diff
@@ -351,13 +351,22 @@ pub struct Options {
     pub src: Option<Vec<String>>,
 
     // Global Formatting options
-    /// The line length to use when enforcing long-lines violations (like
-    /// `E501`). Must be greater than `0` and less than or equal to `320`.
+    /// The line length to use when enforcing long-lines violations (like `E501`)
+    /// and at which the formatter prefers to wrap lines.
+    ///
+    /// The length is determined by the number of characters per line, except for lines containing East Asian characters or emojis.
+    /// For these lines, the [unicode width](https://unicode.org/reports/tr11/) of each character is added up to determine the length.
+    ///
+    /// The value must be greater than `0` and less than or equal to `320`.
+    ///
+    /// Note: The `line-length` isn't a hard upper limit for the formatter.
+    /// It is a way to tell the formatter roughly how long you'd like your lines to be.
+    /// The formatter will both make shorter and longer lines, but generally strives to not exceed the specified length.
```

Comment:
Maybe:

```
/// Note: While the formatter will attempt to format lines such that they remain
/// within the `line-length`, it isn't a hard upper bound, and formatted lines may
/// exceed the `line-length`.
```

Reply:
I like its simplicity. But I would like it to also cover that lines might be shorter (considerably even).